src/elf.rs

```diff
@@ -750,7 +823,7 @@ impl<'a> Elf<'a> {
         header.e_machine = machine.0;
         header.e_type = header::ET_REL;
         header.e_shoff = sh_offset;
-        header.e_shnum = self.nsections;
+        header.e_shnum = if needs_extended_symtab { 0 } else { self.nsections as u16 };
```

Comment:
This may cause an issue too; I believe the value in case of extended table should be `0xFF00` (Need to double check this value) (I forget the elf symbol name, i think it’s `SHN_LORESERVE`, it’s in the elf manpage or the page you linked initially). But pretty sure it’s not 0; that is a valid value, eg no section headers, since they’re technically strippable from elf files, so you can really have 0 section headers :)

Best bet right now is to change this to that value. May just work suddenly :)

Reply:
Oracle's guide says to put 0 here and the Clang generated huge file also has 0 here.

(Don't ask me why the sentinels used by ELF is so inconsistent.)